### ISSUE TITLE
ng9 strictTemplates issues 

### ISSUE THREAD

**Issue by dylanvdmerwe** [NG-ZORRO/ng-zorro-antd#5171]
### Reproduction link
[https://github.com/dylanvdmerwe/zorro-strictTemplates-repo](https://github.com/dylanvdmerwe/zorro-strictTemplates-repo)

### Steps to reproduce
In Angular 9, set `strictTemplates` to true in `angularCompilerOptions` in tsconfig.json. Run `ng build` on the repo.

```
Property 'subscribe' does not exist on type 'EventEmitter<any> | OnClickCallback<any>'.
  Property 'subscribe' does not exist on type 'OnClickCallback<any>'.
```

```
Type 'NzSelectComponent' is missing the following properties from type 'NgModel': control, viewModel, name, isDisabled, and 34 more.
```

Disabling `strictTemplates` allows the code to run as expected. 

The repo attached is just an example of what we have run into. I am sure there are additional components and cases that would need to be checked as well.

Some additional comments here: https://github.com/NG-ZORRO/ng-zorro-antd/issues/4757

Background info: [https://auth0.com/blog/angular-9s-best-hidden-feature-strict-template-checking/](https://auth0.com/blog/angular-9s-best-hidden-feature-strict-template-checking/)

### What is expected?
Ng-Zorro should not prevent the additional checks implemented by `strictTemplates`.

### What is actually happening?
Typing errors prevent the code from compiling. 

| Environment | Info |
|---|---|
| ng-zorro-antd | 9.1.0 |
| Browser | Chrome |

**Comment by vthinkxie** [comment]
plz follow https://github.com/NG-ZORRO/ng-zorro-antd/pull/4994

**Comment by dylanvdmerwe** [comment]
Note that this is still an issue with version `9.1.1`. Repo updated.

**Comment by vthinkxie** [comment]
cc @hsuanxyz

**Comment by hsuanxyz** [comment]
Hi, @dylanvdmerwe 

> Property 'subscribe' does not exist on type 'EventEmitter<any> | OnClickCallback<any>'.
  Property 'subscribe' does not exist on type 'OnClickCallback<any>'.

fix via #5265;

> Type 'NzSelectComponent' is missing the following properties from type 'NgModel': control, viewModel, name, isDisabled, and 34 more.

need to specify the reference name to narrowing types.

```html
<app-test-wrapper [control]="selector">
  <nz-select ngModel="lucy" #selector="ngModel">
                                       ~~~~~~~
  </nz-select>
</app-test-wrapper>
```

**Comment by dylanvdmerwe** [comment]
Thank you for this!
